business/checkers/destinationrules/no_dest_checker.go

```diff
@@ -147,3 +155,74 @@ func (n NoDestinationChecker) hasMatchingService(host kubernetes.Host, itemNames
 	}
 	return false
 }
+
+func (n NoDestinationChecker) isSubsetReferenced(host string, subset string) bool {
+	virtualServices, ok := n.getVirtualServices(host, subset)
+	if ok && len(virtualServices) > 0 {
+		return true
+	}
+
+	return false
+}
+
+func (n NoDestinationChecker) getVirtualServices(virtualServiceHost string, virtualServiceSubset string) ([]kubernetes.IstioObject, bool) {
+	vss := make([]kubernetes.IstioObject, 0, len(n.VirtualServices))
+
+	for _, virtualService := range n.VirtualServices {
+		protocols := [3]string{"http", "tcp", "tls"}
+		for _, protocol := range protocols {
+			specProtocol := virtualService.GetSpec()[protocol]
+			if specProtocol == nil {
+				continue
+			}
+
+			// Getting a []HTTPRoute, []TLSRoute, []TCPRoute
+			slice := reflect.ValueOf(specProtocol)
+			if slice.Kind() != reflect.Slice {
+				continue
+			}
+
+			for routeIdx := 0; routeIdx < slice.Len(); routeIdx++ {
+				httpRoute, ok := slice.Index(routeIdx).Interface().(map[string]interface{})
+				if !ok || httpRoute["route"] == nil {
+					continue
+				}
+
+				// Getting a []DestinationWeight
+				destinationWeights := reflect.ValueOf(httpRoute["route"])
+
+				for destWeightIdx := 0; destWeightIdx < destinationWeights.Len(); destWeightIdx++ {
+					destinationWeight, ok := destinationWeights.Index(destWeightIdx).Interface().(map[string]interface{})
+					if !ok || destinationWeight["destination"] == nil {
+						continue
+					}
+
+					destination, ok := destinationWeight["destination"].(map[string]interface{})
+					if !ok {
+						continue
+					}
+
+					host, ok := destination["host"].(string)
+					if !ok {
+						continue
+					}
+
+					subset, ok := destination["subset"].(string)
+					if !ok {
+						continue
+					}
```

Comment:
Do you think we can extract this method into a common area for different checkers? I see there is a similar code in here: https://github.com/kiali/kiali/blob/3977524605b3bd216d58b8d3093a27f0ae32d352/business/checkers/virtualservices/subset_presence_checker.go#L22-L66

Have you tried it?

Reply:
yes the loop is the same, but the difference is in inner logic, can this logic be provided via local function?

Reply:
can you please send any example of making common functions and where we put it usually?

Reply:
No, I don't know exactly a way to do that. I was thinking if you wanted to get creative with go and find out something.
Let's wait until we find again the same need to reuse this code and then we refactor. So, let's close this comment.

Just for fun, I had a vague idea of having a helper which receives a function as a parameter. This function would have the specific responsibility for each scenario (return VSs or validations or any other interface{}). 
But I guess it is not the time to do so.